src/main/scala/org/constellation/domain/transaction/TransactionChainService.scala

```diff
@@ -50,6 +52,22 @@ class TransactionChainService[F[_]: Concurrent] {
       (snapshotInfo.lastAcceptedTransactionRef, ())
     }
 
+  def checkDummyTransaction(src: String,
+                            dst: String,
+                            amount: Long,
+                            keyPair: KeyPair,
+                            isDummy: Boolean,
+                            fee: Option[Long] = None,
+                            normalized: Boolean = false,
+                            isTest: Boolean = false): F[Transaction] = {
+    val ref = LastTransactionRef.empty
+    val edge: Edge[TransactionEdgeData] = createTransactionEdge(src, dst, ref, amount, keyPair, fee, normalized)
+    val address = edge.observationEdge.parents.head.hash
+    lastTransactionRef.get.map { m =>
```

Comment:
if you're creating a bare new transaction without this `lastTransactionRef`, does it still make sense to do `lastTransactionRef.get` even if it's unused?
imo the same effect you'll have with `Transaction(...).pure[F]`